bzip2-sys/wasm_shim.rs

```diff
@@ -0,0 +1,55 @@
+use std::alloc::{alloc, dealloc, Layout};
+use std::os::raw::{c_int, c_void};
+
+#[no_mangle]
+pub extern "C" fn rust_bzip2_wasm_shim_malloc(size: usize) -> *mut c_void {
+    unsafe {
+        let layout = Layout::from_size_align_unchecked(size, 1);
+        alloc(layout).cast()
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn rust_bzip2_wasm_shim_calloc(nmemb: usize, size: usize) -> *mut c_void {
+    unsafe {
+        let layout = Layout::from_size_align_unchecked(size * nmemb, 1);
+        alloc(layout).cast()
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn rust_bzip2_wasm_shim_free(ptr: *mut c_void) {
+    // layout is not actually used
+    let layout = Layout::from_size_align_unchecked(1, 1);
```

Comment:
This unfortunately is not something I'm personally willing to rely on. The comment that this isn't used is only applicable with the dlmalloc system allocator which can be overridden and may not be true in the future. 

I think that the malloc/free/calloc shims here will need to have a header/footer with the size to reconstruct the `Layout` here.

Reply:
Do you think that plays a role for WASM though? Can you even use `dlmalloc` with WASM? I kinda copied this part - would have to dig deeper into the `dealloc` implementation.